input/input.go

```diff
@@ -67,15 +67,18 @@ func (in DefaultHandler) Process(metric *schema.MetricData, partition int32) {
 		return
 	}
 
+	schemaI, _ := mdata.MatchSchema(metric.Name)
```

Comment:
> i think it would make sense to have idx.AddOrgUpdate() to return idx.Node

in the case where the metric already exists, we can currently just check for the def in DefById via a simple id lookup, and return if it exists. To support what you suggest, I think we would have either do a lookup in the tree in that case, or somehow link to the idx.Node, with extra data in DefsById or something.
I think It's not the index' concern that the caller of the index would like to know these properties about an entry when it tries to add a new entry, I rather keep the index focused on it's task of being an index rather than trying to accommodate for this.
the current caching mechanism seems to work well enough.